src/Workspaces/CSharp/Portable/CodeGeneration/ParameterGenerator.cs

```diff
@@ -82,10 +82,11 @@ internal static ParameterSyntax GetParameter(IParameterSymbol p, CodeGenerationO
                 return reusableSyntax;
             }
 
+            var nullableAnnotation = p.NullableAnnotation;
             return SyntaxFactory.Parameter(p.Name.ToIdentifierToken())
-                    .WithAttributeLists(GenerateAttributes(p, isExplicit, options))
+                    .WithAttributeLists(GenerateAttributes(p, isExplicit, options, ref nullableAnnotation))
```

Comment:
Can we avoid the ref parameter here and just have this return a tuple or something? It's kinda terrifying to me that the With* methods here aren't reorderable because of side effects.

Reply:
Anyway the original annotation needs to be passed as input.

Reply:
Fixed.